Sources/PubNub/Helpers/Crypto/Miscellaneous/CryptoInputStream.swift

```diff
@@ -26,11 +26,30 @@
 //
 
 import Foundation
+import CommonCrypto
 
 /// A stream that provides read-only stream functionality while performing crypto operations
 public class CryptoInputStream: InputStream {
   // swiftlint:disable:previous type_body_length
-
+  public struct DataSource {
+    let key: Data
+    let iv: Data
+    let options: CCOptions
+    let cipher: Cipher
+  }
+  public struct Cipher {
+    let algorithm: CCAlgorithm
+    let blockSize: Int
+    
+    func outputSize(from inputBytes: Int) -> Int {
+      inputBytes + (blockSize - (inputBytes % blockSize))
```

Comment:
CCCrypto actually can tell you what resulting size you will get (depending on from operation type and source data size).